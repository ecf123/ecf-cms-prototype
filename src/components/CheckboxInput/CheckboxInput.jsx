import { useState } from "react";
import "./CheckboxInput.scss";

const CheckboxInput = ({handleClick, handleInput}) => {

  return (
    <div className="checkbox">
      <input onChange={handleInput} type="text" className="checkbox__input" />
      <input onClick={handleClick} type="checkbox" className="checkbox__selector" />
    </div>
  );
};

export default CheckboxInput;
