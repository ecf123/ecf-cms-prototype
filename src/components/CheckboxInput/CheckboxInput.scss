@use "../../styles/variables.scss" as *;

.checkbox {
  margin: 100px;
  border: 1px solid #d0d5dd;
  max-width: 512px;
  height: 48px;
  border-radius: 8px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 15px;

  &__input {
    border: none;
    width: 450px;
    height: 80%;
    font-family: $font-secondary;
    font-size: $font-size-small;
    color: $color-grey-normal;
  }

  &__input:focus {
    outline: none;
  }
}
