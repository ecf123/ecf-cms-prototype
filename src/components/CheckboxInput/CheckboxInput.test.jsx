import { fireEvent, render, screen } from "@testing-library/react";
import { customRender } from "../../utils/testUtils";
import CheckboxInput from "./CheckboxInput.jsx";

it("Should match the snapshot", () => {
  const { container } = customRender(<CheckboxInput />);

  expect(container).toMatchSnapshot();
});

it("Should render the Checkbox Input textarea", () => {
  // 1. Arrange
  render(<CheckboxInput />);
  // 2. Act
  const textArea = screen.getByRole("textbox");
  // 3. Assert
  expect(textArea).toBeInTheDocument();
  //comment
});

it("Should render the Checkbox has not been clicked on load", () => {
  render(<CheckboxInput />);
  const checkbox = screen.getByRole("checkbox");
  expect(checkbox).not.toBeChecked();
});

it("Should render the Checkbox has been clicked by user", () => {
  render(<CheckboxInput />);
  const checkbox = screen.getByRole("checkbox");
  fireEvent.click(checkbox);
  expect(checkbox).toBeChecked();
});
