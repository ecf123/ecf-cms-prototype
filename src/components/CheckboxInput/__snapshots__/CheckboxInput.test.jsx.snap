// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
<div>
  <div
    class="checkbox"
  >
    <input
      class="checkbox__input"
      type="text"
    />
    <input
      type="checkbox"
    />
  </div>
</div>
`;
